pixels/endpoints/moderation.py

```diff
@@ -0,0 +1,188 @@
+from httpx import AsyncClient
+
+from pixels.constants import Discord, Server, Sizes
+from pixels.models import Message, ModBan, PixelHistory, User
+from pixels.utils import auth
+
+log = logging.getLogger(__name__)
+router = APIRouter(
+    tags=["Moderation Endpoints"],
+    include_in_schema=Server.SHOW_DEV_ENDPOINTS,
+    dependencies=[Depends(auth.JWTBearer(is_mod_endpoint=True))]
+)
+
+
+@router.get("/mod", response_model=Message)
+async def mod_check(request: Request) -> Message:
+    """Check if the authenticated user is a mod."""
+    return Message(message="Hello fellow moderator!")
+
+
+@router.post("/set_mod", response_model=Message)
+async def set_mod(request: Request, user: User) -> Message:
+    """Make another user a mod."""
+    user_id = user.user_id
+    conn = request.state.db_conn
+    async with conn.transaction():
+        user_state = await conn.fetchrow("SELECT is_mod FROM users WHERE user_id = $1", user_id)
+        if user_state is None:
+            return Message(message=f"User with user_id {user_id} does not exist.")
+        elif user_state['is_mod']:
+            return Message(message=f"User with user_id {user_id} is already a mod.")
+
+        await conn.execute("UPDATE users SET is_mod = true WHERE user_id = $1", user_id)
+    return Message(message=f"Successfully set user with user_id {user_id} to mod")
+
+
+@router.post("/mod_ban", response_model=ModBan)
+async def ban_users(request: Request, user_list: list[User]) -> ModBan:
+    """Ban users from using the API."""
+    conn = request.state.db_conn
+    users = [user.user_id for user in user_list]
+
+    records = await conn.fetch("SELECT * FROM users WHERE user_id=any($1::bigint[])", tuple(users))
+    db_users = [record["user_id"] for record in records]
+
+    non_db_users = set(users) - set(db_users)
+
+    # Ref:
+    # https://magicstack.github.io/asyncpg/current/faq.html#why-do-i-get-postgressyntaxerror-when-using-expression-in-1
+    await conn.execute("UPDATE users SET is_banned=TRUE WHERE user_id=any($1::bigint[])", db_users)
+
+    await conn.execute("UPDATE pixel_history SET deleted=TRUE WHERE user_id=any($1::bigint[])", db_users)
+
+    await request.state.canvas.sync_cache(conn, skip_check=True)
+
+    return ModBan(banned=db_users, not_found=list(non_db_users))
+
+
+@router.get("/pixel_history", response_model=t.Union[PixelHistory, Message])
+async def pixel_history(
+        request: Request,
+        x: int = Sizes.X_QUERY_VALIDATOR,
+        y: int = Sizes.Y_QUERY_VALIDATOR
+) -> t.Union[PixelHistory, Message]:
+    """Get the user who placed a specific pixel given its coordinates."""
+    sql = (
+        "SELECT user_id::text FROM pixel_history "
+        "WHERE x=$1 AND y=$2 AND NOT deleted "
+        "ORDER BY pixel_history_id DESC "
+        "LIMIT 1"
+    )
+    record = await request.state.db_conn.fetchrow(sql, x, y)
+
+    if not record:
+        return Message(message=f"No user history for pixel ({x}, {y})")
```

Comment:
I think the other messages ended with a dot.
```suggestion
        return Message(message=f"No user history for pixel ({x}, {y}).")
```